Mathlib/Order/WithBot.lean

```diff
@@ -168,17 +169,24 @@ theorem eq_unbot_iff {a : α} {b : WithBot α} (h : b ≠ ⊥) :
 
 section LE
 
-variable [LE α]
+variable [LE α] {x y : WithBot α}
 
-instance (priority := 10) le : LE (WithBot α) :=
-  ⟨fun o₁ o₂ => ∀ a : α, o₁ = ↑a → ∃ b : α, o₂ = ↑b ∧ a ≤ b⟩
+/-- The order on `WithBot α`, defined by `⊥ ≤ ⊥`, `⊥ ≤ ↑a` and `a ≤ b → ↑a ≤ ↑b`. -/
+@[mk_iff]
+inductive le : WithBot α → WithBot α → Prop
```

Comment:
How does this compare to
```lean
def le : WithBot α → WithBot α → Prop
 | (a : WithBot α), (b: WithBot α) => a ≤ b
 | none, _ => True
```

Reply:
This suggestion is invalid since you aren't covering the `(a : alpha), \bot` case (at least once you fix the typo in the first branch)

Once you've covered it, the following won't work:
```
example : \forall x y : WithBot alpha, x <= y -> True
  | \bot, \bot, _ => sorry
  | \bot, (b : alpha), _ => sorry
  | (a : alpha), (b : alpha) => sorry
```